components/experimental/src/dimension/percent/format.rs

```diff
@@ -0,0 +1,299 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use fixed_decimal::{FixedDecimal, Sign};
+
+use writeable::Writeable;
+
+use crate::dimension::provider::percent::PercentEssentialsV1;
+
+use super::options::{Display, PercentFormatterOptions};
+
+pub struct FormattedPercent<'l> {
+    pub(crate) value: &'l FixedDecimal,
+    pub(crate) essential: &'l PercentEssentialsV1<'l>,
+    pub(crate) options: &'l PercentFormatterOptions,
+}
+
+impl<'l> Writeable for FormattedPercent<'l> {
+    fn write_to<W>(&self, sink: &mut W) -> core::result::Result<(), core::fmt::Error>
+    where
+        W: core::fmt::Write + ?Sized,
+    {
+        match self.options.display {
+            // In the Standard display, we take the
+            Display::Standard => self
+                .essential
+                .positive_pattern
+                .interpolate([self.value])
+                .write_to(sink)?,
+            Display::Approximate => {
+                // Removing the sign from the value
+                let abs_value = match self.value.sign() {
+                    Sign::Negative => self.value.clone().with_sign(Sign::None),
+                    _ => self.value.to_owned(),
+                }
+                .to_string();
+
+                // The approximate sign gets pre-pended
+                let sign = match self.value.sign() {
+                    Sign::Negative => {
+                        String::new()
+                            + &self.essential.approximately_sign
+                            + &self.essential.minus_sign
+                    }
+                    _ => self.essential.approximately_sign.to_string(),
+                };
+
+                self.essential
+                    .negative_pattern
+                    .interpolate([abs_value, sign])
+                    .write_to(sink)?;
+            }
+            // TODO: Determine if we throw an error when explicit plus is called and the value is negative.
+            Display::ExplicitPlus => self
+                .essential
+                .negative_pattern
+                .interpolate([self.value.to_string(), self.essential.plus_sign.to_string()])
+                .write_to(sink)?,
+        };
```

Comment:
Unsure what to do for this portion. There's one of 2 assumptions we could make here.

1. They want an absolute value. Example:
  a. Input: `-123` -> output `+123%`

2. They just want any visible sign (plus or minus).
  a. Input: `-123` -> output `-123%`
  b. Input: `123` -> output `+123%`

I think my assumption would be 2.